base-containers/base/bin/inginious-ipython

```diff
@@ -0,0 +1,24 @@
+#! /usr/bin/python3
+# coding=utf-8
+
+import sys
+import IPython.core.interactiveshell
+
+if len(sys.argv) == 1:
+    print("No file given", file=sys.stderr)
+    exit(1)
+
+# A small hack to force IPython to interpret the given file as a IPYthon file.
+# By default, it only parses correctly files with the .ipy extension.
+
+shell = IPython.core.interactiveshell.InteractiveShell()
+sys.argv = sys.argv[1:]  # correct the args
+
+# Add the utility function to the global namespace
+shell.run_cell("from inginious_container_api.input import *", store_history=False)
+shell.run_cell("from inginious_container_api.feedback import *", store_history=False)
+shell.run_cell("from inginious_container_api.lang import *", store_history=False)
+shell.run_cell("from inginious_container_api.rst import *", store_history=False)
+shell.run_cell("from inginious_container_api.run_student import *", store_history=False)
+
+shell.safe_execfile_ipy(sys.argv[0])
```

Comment:
I guess that this would prevent running a ipython script with several arguments from a Bash run file ?

Reply:
It doesn't. Line 15 correct the args, such that the script see the correct ones. For example, if you run `/bin/inginious-ipython myscript.py something somethingelse`, then `myscript.py` will see the following as `sys.argv`:
`['myscript.py', 'something', 'somethingelse']`